### PR TITLE
add weibo/user-bookmarks 微博用户收藏动态 route

### PR DESCRIPTION
**Pull request by cztchoice**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/weibo/user_bookmarks/1195230310
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/weibo/user-bookmarks.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['weibo.com/'],
+            target: '/friends',
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:33]: This should be the same or similar to `path`

File: lib/routes/weibo/user-bookmarks.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        } else {
+            const routeParams = querystring.parse(ctx.req.param('routeParams'));
+            displayVideo = fallback(undefined, queryToBoolean(routeParams.displayVideo), true) ? '1' : '0';
+            displayArticle = fallback(undefined, queryToBoolean(routeParams.displayArticle), false) ? '1' : '0';
+            displayComments = fallback(undefined, queryToBoolean(routeParams.displayComments), false) ? '1' : '0';
+        }
+    }
+
+    const uid = await cache.tryGet(
+        `weibo:user_bookmarks:login-user`,
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:69]:
```suggestion
        'weibo:user_bookmarks:login-user',
```

File: lib/routes/weibo/user-bookmarks.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        async () => {
+            const _r = await got({
+                method: 'get',
+                url: 'https://m.weibo.cn/api/config',
+                headers: {
+                    Referer: `https://m.weibo.cn/`,
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:75]:
```suggestion
                    Referer: 'https://m.weibo.cn/',
```

File: lib/routes/weibo/user-bookmarks.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    const bookmarkContainerId = params.get('lfid');
+
+    const cards = await cache.tryGet(
+        `weibo:user_bookmarks:cards:${uid}`,
+        async () => {
+            const _r = await got({
+                method: 'get',
+                url: `https://m.weibo.cn/api/container/getIndex?containerid=${bookmarkContainerId}&openApp=0`,
+                headers: {
+                    Referer: `https://m.weibo.cn/`,
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:120]:
```suggestion
                    Referer: 'https://m.weibo.cn/',
```

File: lib/routes/weibo/user-bookmarks.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        false
+    );
+    const resultItems = await Promise.all(
+        cards
+            .filter((item) => item.mblog)
+            .map(async (item) => {
+                const key = 'weibo:user_bookmarks:' + item.mblog.bid;
+                const data = await cache.tryGet(key, () => weiboUtils.getShowData(uid, item.mblog.bid));
+
+                if (data && data.text) {
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:138]:
```suggestion
                if (data?.text) {
```

File: lib/routes/weibo/user-bookmarks.ts [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                    if (item.mblog.retweeted_status && data.retweeted_status) {
+                        item.mblog.retweeted_status.created_at = data.retweeted_status.created_at;
+                    }
+                } else {
+                    item.mblog.created_at = timezone(item.mblog.created_at, +8);
+                }
+
+                // 转发的长微博处理
+                const retweet = item.mblog.retweeted_status;
+                if (retweet && retweet.isLongText) {
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:151]:
```suggestion
                if (retweet?.isLongText) {
```

File: lib/routes/weibo/user-bookmarks.ts (hunk 3 above)
Comment (cztchoice) [reply lib/routes/weibo/user-bookmarks.ts]: fixed

File: lib/routes/weibo/user-bookmarks.ts (hunk 4 above)
Comment (cztchoice) [reply lib/routes/weibo/user-bookmarks.ts]: fixed

File: lib/routes/weibo/user-bookmarks.ts (hunk 6 above)
Comment (cztchoice) [reply lib/routes/weibo/user-bookmarks.ts]: fixed

File: lib/routes/weibo/user-bookmarks.ts (hunk 1 above)
Comment (cztchoice) [reply lib/routes/weibo/user-bookmarks.ts]: fixed

File: lib/routes/weibo/user-bookmarks.ts (hunk 2 above)
Comment (cztchoice) [reply lib/routes/weibo/user-bookmarks.ts]: fixed

File: lib/routes/weibo/user-bookmarks.ts [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['weibo.com/'],
+            target: '/weibo/user_bookmarks/:uid',
```
Comment (TonyRL) [inline lib/routes/weibo/user-bookmarks.ts:33]:
```suggestion
            target: '/user_bookmarks/:uid',
```

**Pull request merged**